src/main/java/com/rybka/dao/CurrencyHistoryDAO.java

```diff
@@ -28,7 +29,20 @@ public void save(CurrencyHistory currencyHistory) {
 
     public List<CurrencyHistory> findAll() {
         try (Session session = HibernateUtil.getSessionFactory().openSession()) {
-            return session.createQuery("from CurrencyHistory", CurrencyHistory.class).list();
+            return session.createQuery(QueryConstants.SELECT_QUERY, CurrencyHistory.class).list();
+        } catch (Exception e) {
+            transaction.rollback();
+            log.error("Error while retrieving data from DB. Reason: " + e.getMessage());
+            throw new DBConnectionException("Unable to access to DB. Reason: " + e.getMessage());
+        }
+    }
+
+    public List<CurrencyHistory> findFiveRecords() {
```

Comment:
findAllLimit(int count) would be better and reusable.

Reply:
fixed